gateway/src/node/client_handling/websocket/connection_handler.rs

```diff
@@ -381,29 +422,42 @@ where
                 return ServerResponse::new_error(e.to_string());
             }
         };
-        if credential.verify(&self.aggregated_verification_key) {
-            match Bandwidth::try_from(credential) {
-                Ok(bandwidth) => {
-                    if let Err(e) = Bandwidth::increase_bandwidth(
-                        &self.bandwidths,
-                        &self.remote_address.unwrap(),
-                        bandwidth.value(),
-                    )
+
+        if !credential.verify(&self.aggregated_verification_key) {
+            return ServerResponse::Bandwidth { status: false };
+        }
+
+        match Bandwidth::try_from(credential) {
+            Ok(bandwidth) => {
+                let mut bandwidth_value = bandwidth.value();
+                if bandwidth_value > i64::MAX as u64 {
+                    // note that this would have represented more than 1 exabyte,
+                    // which is like 125,000 worth of hard drives so I don't think we have
+                    // to worry about it for now...
+                    warn!("Somehow we received bandwidth value higher than 9223372036854775807. Going to cap it at that amount.");
+                    bandwidth_value = i64::MAX as u64;
```

Comment:
Even though this is a strange edge case, I don't think a warning and capping the value is the right approach here, as the client might have genuinely bought that much bandwidth, so we're just cutting down value here. We should return an error and not increase the GW bandwidth until the client sends an appropriate value.

Reply:
I made the comment on the first appearance of this code, and I think it's still viable, but the code was moved somewhere else in another commit

Reply:
Personally I don't think client could have genuinely brought this much bandwidth - it's over 1 exabyte which is around 1% capacity of ALL datacenters in the world : ) But perhaps returning an error as you suggest might be a better approach than capping it - I'm going to change it.